dataflux_pytorch/tests/test_dataflux_mapstyle_dataset.py

```diff
@@ -97,6 +98,67 @@ def test_init_with_required_parameters(self, mock_dataflux_core):
             f"got listed objects {ds.objects}, want {self.want_objects}",
         )
 
+    @mock.patch("dataflux_pytorch.dataflux_mapstyle_dataset.dataflux_core")
+    def test_init_without_storage_client(self, mock_dataflux_core):
+        """Tests the DataFluxMapStyleDataset can be initiated without storage_client."""
+        # Arrange.
+        mock_listing_controller = mock.Mock()
+        mock_listing_controller.run.return_value = self.want_objects
+        mock_dataflux_core.fast_list.ListingController.return_value = (
+            mock_listing_controller)
+
+        # Act.
+        ds = dataflux_mapstyle_dataset.DataFluxMapStyleDataset(
+            project_name=self.project_name,
+            bucket_name=self.bucket_name,
+            # Ensure the dataset can be constructed without setting storage_client.
+            # storage_client=self.storage_client,
+        )
+
+        # Assert.
+        self.assertEqual(
+            ds.objects,
+            self.want_objects,
+            f"got listed objects {ds.objects}, want {self.want_objects}",
+        )
+
+        # Ensure the dataset can be constructed without setting storage_client (#58).
+        self.assertIsNone(
+            ds.storage_client,
+            "storage_client was unexpectedly constructed on init.")
+
+    @mock.patch("dataflux_pytorch.dataflux_mapstyle_dataset.dataflux_core")
+    def test_init_without_storage_client_constructed_when_needed(
+            self, mock_dataflux_core):
+        """Tests the DataFluxMapStyleDataset can be initiated without storage_client."""
+        # Arrange.
+        mock_listing_controller = mock.Mock()
+        mock_listing_controller.run.return_value = self.want_objects
+        mock_dataflux_core.fast_list.ListingController.return_value = (
+            mock_listing_controller)
+
+        # Act.
+        ds = dataflux_mapstyle_dataset.DataFluxMapStyleDataset(
+            project_name=self.project_name,
+            bucket_name=self.bucket_name,
+            # Ensure the dataset can be constructed without setting storage_client.
+            # storage_client=self.storage_client,
```

Comment:
Same as above ? (I dont think we need the this commented line 

`storage_client = self.storage_client`